tsconfig.json

```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "https://json.schemastore.org/tsconfig",
```

Comment:
I've never used $schema in my configs. Do we really need it?
Also, do we need all the other settings? please go over the list and see if we can omit uneeded settings.

Reply:
we can check the settings, maybe in a separate issue. I needed a root tsconfig to make knip happy so took all the common config items and extracted out

Reply:
I removed the schema entry, looks like IDEs like VSCode still know what properties are allowed and can provide intellisense